File: frontend/lib/model/Candidacy.dart
enum StateCandidacy { WAITING, POSITIVE, NEGATIVE }

class Candidacy {
  String _id;
  String _designer;
  String _projectProposer;
  String _project;
  String _dateOfCandidacy;
  StateCandidacy _state;
  String _dateOfExpire;
  String _message;

  Candidacy.complete(
      this._id,
      this._designer,
      this._projectProposer,
      this._project,
      this._dateOfCandidacy,
      this._state,
      this._dateOfExpire,
      this._message);

  Candidacy();

  String getId() {
    return this._id;
  }

  void setId(String id) {
    _id = id;
  }

  String getDesigner() {
    return this._designer;
  }

  void setDesigner(String designer) {
    _designer = designer;
  }

  String getProjectProposer() {
    return this._projectProposer;
  }

  void setProjectProposer(String projectProposer) {
    _projectProposer = projectProposer;
  }

  String getProject() {
    return this._project;
  }

  void setProject(String project) {
    _project = project;
  }

  String getDateOfCandidacy() {
    return this._dateOfCandidacy;
  }

  void setDateOfCandidacy(String dateOfCandidacy) {
    _dateOfCandidacy = dateOfCandidacy;
  }

  StateCandidacy getState() {
    return this._state;
  }

  void setState(StateCandidacy state) {
    _state = state;
  }

  String getDateOfExpire() {
    return this._dateOfExpire;
  }

  void setDateOfExpire(String dateOfExpire) {
    _dateOfExpire = dateOfExpire;
  }

  String getMessage() {
    return this._message;
  }

  void setMessage(String message) {
    _message = message;
  }
}

File: backend/src/main/java/it/unicam/qwert123/doit/backend/models/Candidacy.java
package it.unicam.qwert123.doit.backend.models;

import java.util.Date;
import java.util.UUID;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Document(collection = "candidacy")
public class Candidacy {
    @Id
    private UUID id;
    private String designer;
    private String projectProposer;
    private UUID project;
    private Date dateOfCandidacy;
    private StateCandidacy state;
    private Date dateOfExpire;
    private String message;

    public enum StateCandidacy  {
        WAITING, POSITIVE, NEGATIVE
    } 
    
}
